### ISSUE TITLE
Testing with Jest Timeout Issue

### ISSUE THREAD

**Issue by fnoori** [socketio/socket.io#4633]
**Describe the bug**
Going through the testing page: https://socket.io/docs/v4/testing/, under **Example with `jest`**.

When I go to run the tests, I get a timeout error:
```
Error: thrown: "Exceeded timeout of 5000 ms for a hook.
Use jest.setTimeout(newTimeout) to increase the timeout value, if this is a long-running test."
```

**To Reproduce**

_disclaimer_, I am using Webstorm to run this. I can confirm that my Webstorm configurations is not the issue, I am able to run all other unit tests successfully. 

1. Go to: https://socket.io/docs/v4/testing/
2. Copy/Paste the code snippet under **Example with `jest`**, onto your local machine
3. Run the unit test
4. Expect to see error outlined above

Socket.IO server version: `4.5.2`

*Server*

```js
import {createServer} from 'http'
import {io as Client} from 'socket.io-client'
import {Server} from 'socket.io'

describe('my awesome project', () => {
  let io, serverSocket, clientSocket

  beforeAll((done) => {
    const httpServer = createServer()
    io = new Server(httpServer)
    httpServer.listen(() => {
      const port = httpServer.address().port
      clientSocket = new Client(`http://localhost:${port}`)
      io.on('connection', (socket) => {
        serverSocket = socket
      })
      clientSocket.on('connect', done)
    })
  })

  afterAll(() => {
    io.close()
    clientSocket.close()
  })

  test('should work', (done) => {
    clientSocket.on('hello', (arg) => {
      expect(arg).toBe('world')
      done()
    })
    serverSocket.emit('hello', 'world')
  })

  test('should work (with ack)', (done) => {
    serverSocket.on('hi', (cb) => {
      cb('hola')
    })
    clientSocket.emit('hi', (arg) => {
      expect(arg).toBe('hola')
      done()
    })
  })
})
```

*Client*

**No Client**

**Expected behaviour**
Test should run successfully and not get a timeout error

**Platform:**
 - Device: MacBook 13-inch, M1, 2020
 - OS: MacOS 13.2.1 (22D68)
 - Node: v18.12.1

**Additional context**
- My suspicion is that the error is occurring because of `clientSocket.on('connect', done)`, where the `done` callback is not called
- I'm able to put a breakpoint on `clientSocket = new Client(http://localhost:${port})` and I can see that `clientSocket` is in fact defined

**Comment by darrachequesne** [comment]
Hi! I was not able to reproduce: https://github.com/socketio/socket.io-fiddle/tree/jest

Could you please check?

**Comment by fnoori** [comment]
> Hi! I was not able to reproduce: https://github.com/socketio/socket.io-fiddle/tree/jest
> 
> Could you please check?

Thanks for the reply, @darrachequesne, I've found the source to my problem!

So turns out my problem was that I was using the incorrect jest package. I was using the jest package that came pre-packaged with `.../node_modules/react-scripts`, so when I updated the jest package in my IDE's configurations to use `.../node_modules/jest`, my tests passed!

When I ran the unit test from the repo you commented, it worked out of the box, my assumption is it worked right away because there is no mention of React in that repo (no chance of my IDE using `.../node_modules/react-scripts`). Whereas, in my repo I have a React dependency in my `package.json`
